src/PSAppDeployToolkit/Public/Set-ADTIniValue.ps1

```diff
@@ -63,11 +63,23 @@ function Set-ADTIniValue
         [System.String]$FilePath,
 
         [Parameter(Mandatory = $true)]
-        [ValidateNotNullOrEmpty()]
+        [ValidateScript({
+                if ([string]::IsNullOrWhiteSpace($_))
+                {
+                    $PSCmdlet.ThrowTerminatingError((New-ADTValidateScriptErrorRecord -ParameterName Path -ProvidedValue $_ -ExceptionMessage 'The specified section cannot be null, empty, or whitespace.'))
+                }
+                return $true
+            })]
         [System.String]$Section,
 
         [Parameter(Mandatory = $true)]
-        [ValidateNotNullOrEmpty()]
+        [ValidateScript({
+                if ([string]::IsNullOrWhiteSpace($_))
```

Comment:
You're using a mix of `[string]` and `[System.String]`. While I prefer the latter and it's the more common vernacular within the module, whatever you choose at least be consistent.